lms: init at 3.51.1

mksafavi opened the pull request:
## Description of changes

This supersedes https://github.com/NixOS/nixpkgs/pull/147408
I've packaged the core functionality of it. There are a few places that I think I didn't do it correctly. I marked them with `FIXME`. When they're taken care of, I'll add the config options and the service for it.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux (It should support arm64 but I don't have nix setup on my arm system yet)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mksafavi committed:
lms: init at 3.51.1

Aleksanaa commented on pkgs/by-name/lm/lms/package.nix:
```diff
@@ -0,0 +1,88 @@
+  meta = with lib; {
+    homepage = "https://github.com/epoupon/lms";
+    description = "Lightweight Music Server - Access your self-hosted music using a web interface";
+    license = licenses.gpl3Plus; # FIXME: gpl3Plus or gpl3Only?
```
gpl3Plus, see https://github.com/epoupon/lms/blob/4e5a742441ce594209e2be4d79ffca35a95a8b0c/src/lms/main.cpp#L9

Aleksanaa commented on pkgs/by-name/lm/lms/package.nix:
```diff
@@ -0,0 +1,88 @@
+    homepage = "https://github.com/epoupon/lms";
+    description = "Lightweight Music Server - Access your self-hosted music using a web interface";
+    license = licenses.gpl3Plus; # FIXME: gpl3Plus or gpl3Only?
+    platforms = with platforms; [
```
`platforms.linux` is mostly okay.

Aleksanaa commented on pkgs/by-name/lm/lms/package.nix:
```diff
@@ -0,0 +1,88 @@
+    sha256 = "sha256-5lEbrB218EVVHIzo1efvQYybut2OpfDKpLlRs0brhXg=";
+  };
+
+  nativeBuildInputs = [
```
Please enable `strictDeps = true`. Some of these should go to `buildInputs`

Aleksanaa commented on pkgs/by-name/lm/lms/package.nix:
```diff
@@ -0,0 +1,88 @@
+
+  # FIXME: Should I replace these with a git-patch?
+  postPatch = ''
+    substituteInPlace src/lms/main.cpp --replace '"/etc/lms.conf"' '"${placeholder "out"}/share/lms/lms.conf"'
```
```suggestion
    substituteInPlace src/lms/main.cpp --replace '"/etc/lms.conf"' "$out/share/lms/lms.conf"
```

Aleksanaa commented on pkgs/by-name/lm/lms/package.nix:
```diff
@@ -0,0 +1,88 @@
+  cmakeFlags = [ "-DCMAKE_BUILD_TYPE=Release" ];
+
+  postInstall = ''
+    substituteInPlace $out/share/lms/lms.conf --replace \
```
```suggestion
    substituteInPlace $out/share/lms/lms.conf --replace-fail \
```

Also for others.

Aleksanaa commented on pkgs/by-name/lm/lms/package.nix:
```diff
@@ -0,0 +1,88 @@
+    libSM
+    libICE
+  ];
+  # FIXME: Is there any way to add libSM libICE into this without using gappsWrapper?
```
Not a gtk app? Please use `wrapProgram` in `makeWrapper`.

mksafavi commented:
Hi @Aleksanaa.
Thanks for reviewing this.
I applied the suggestions you made. 
Let me know if there's anything else.

marcusramberg commented:
Update commits should be squashed into the original one.

mksafavi commented:
I have two issues with this.
1. After installing it, bin files aren't available in shell. I think I'm missing something in `wrapProgram`. 
2. The systemd service doesn't work. It's missing the `ExecStart=` line.

marcusramberg reviewed (approved):
lgtm.

```
1 package built:
lms
```

Started and tested that you can log into the webui.

mksafavi commented on pkgs/by-name/lm/lms/package.nix:
```diff
@@ -0,0 +1,85 @@
+
+  cmakeFlags = [ "-DCMAKE_BUILD_TYPE=Release" ];
+
+  preFixup = ''
+    wrapProgram $out/bin/lms \
+      --prefix LD_LIBRARY_PATH : "${lib.strings.makeLibraryPath buildInputs}"
+    wrapProgram $out/bin/lms-metadata \
+      --prefix LD_LIBRARY_PATH : "${lib.strings.makeLibraryPath buildInputs}"
+    wrapProgram $out/bin/lms-recommendation \
+      --prefix LD_LIBRARY_PATH : "${lib.strings.makeLibraryPath buildInputs}"
+  '';
```
Here I'm adding all the `buildInputs` into the `lib.strings.makeLibraryPath`. It's creating a lot of `LD_LIBRARY_PATH` exports. I think only `libSM` and `libICE` need to be in the `LD_LIBRARY_PATH`.
Should I fix it?

marcusramberg commented on pkgs/by-name/lm/lms/package.nix:
```diff
@@ -0,0 +1,85 @@
+    owner = "epoupon";
+    repo = "lms";
+    rev = "v${version}";
+    sha256 = "sha256-5lEbrB218EVVHIzo1efvQYybut2OpfDKpLlRs0brhXg=";
```
```suggestion
    hash = "sha256-5lEbrB218EVVHIzo1efvQYybut2OpfDKpLlRs0brhXg=";
```

mksafavi commented:
I think this is ready.
I fixed the systemd service issue. the original service file was `default.service`. I renamed it to `lsmd.service`
Although there's something that I don't understand. 
Why does nix create an extra malformed service file? 
There's another service file `lms.service` that I didn't create. I doesn't work because it doesn't have `ExecStart` line.

Is there anything wrong with the way I'm packaging it that creates this extra file? 
I also noticed this on another program that I use called `lact`. It has two service files. `lactd.service` that runs the daemon and `lact.service` which is a broken file.

mksafavi commented:
@Aleksanaa 
Is there anything else to do to merge this?
